image/helpers.go

```diff
@@ -221,15 +221,18 @@ func (tsto *ToolingStore) DownloadSnap(name string, revision snap.Revision, opts
 		targetDir = pwd
 	}
 
-	spec := store.SnapSpec{
+	actions := []*store.SnapAction{{
+		Action:   "download",
 		Name:     name,
-		Channel:  opts.Channel,
 		Revision: revision,
-	}
-	snap, err := sto.SnapInfo(spec, tsto.user)
+		Channel:  opts.Channel,
```

Comment:
we need code like:  
https://github.com/snapcore/snapd/blob/master/overlord/snapstate/storehelpers.go#L85

or do something in store/ itself as well